grakn-graql/src/main/java/ai/grakn/graql/internal/pattern/property/AbstractIsaProperty.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * Grakn - A Distributed Semantic Database
+ * Copyright (C) 2016-2018 Grakn Labs Limited
+ *
+ * Grakn is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Grakn is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Grakn. If not, see <http://www.gnu.org/licenses/gpl.txt>.
+ */
+
+package ai.grakn.graql.internal.pattern.property;
+
+import ai.grakn.GraknTx;
+import ai.grakn.concept.SchemaConcept;
+import ai.grakn.concept.Type;
+import ai.grakn.exception.GraqlQueryException;
+import ai.grakn.graql.Var;
+import ai.grakn.graql.admin.UniqueVarProperty;
+import ai.grakn.graql.admin.VarPatternAdmin;
+import com.google.common.collect.ImmutableSet;
+
+import java.util.Collection;
+import java.util.stream.Stream;
+
+/**
+ * @author Jason Liu
+ */
+
+abstract class AbstractIsaProperty extends AbstractVarProperty implements UniqueVarProperty, NamedProperty {
```

Comment:
I always make mistakes and override the wrong things when I make abstract classes. So I label all the methods I don't want to override as `final`:

```
public final String getProperty() {
    return type().getPrintableName();
}
```

Maybe do that here for ones you don't want to override?